File: frontend/src/features/event/model/split_time_analysis.ts
import type { ControlSegment } from './control_segment'
import type { ControlSequenceSegment } from './control_sequence_segment'

export interface SplitTimeAnalysis {
    resultListId: number
    eventId: number
    classResultShortName: string
    controlSegments: ControlSegment[]
    sequenceSegments: ControlSequenceSegment[]
}

File: backend/src/main/java/de/jobst/resulter/domain/analysis/SplitTimeAnalysis.java
package de.jobst.resulter.domain.analysis;

import de.jobst.resulter.domain.ClassResultShortName;
import de.jobst.resulter.domain.EventId;
import de.jobst.resulter.domain.ResultListId;
import org.jmolecules.ddd.annotation.ValueObject;

import java.util.List;

@ValueObject
public record SplitTimeAnalysis(ResultListId resultListId, EventId eventId, ClassResultShortName classResultShortName,
                                List<ControlSegment> controlSegments) {
}
